riscv/debug_module.cc

```diff
@@ -522,6 +536,10 @@ bool debug_module_t::dmi_read(unsigned address, uint32_t *value)
       case DM_DMCS2:
         result = set_field(result, DM_DMCS2_GROUP, selected_hart_state().haltgroup);
         break;
+      case DM_CUSTOM:
+        result = set_field(result, 1, hart_available_state[0]);
+        result = set_field(result, 2, hart_available_state[1]);
```

Comment:
Similarly, it would be cleaner if the hardcoded array size of 2 weren't duplicated here; could use a loop that iterates until `sizeof(hart_available_state) / sizeof(*hart_available_state)` instead.